app/controllers/UploadWrittenAuthorisationController.scala

```diff
@@ -95,12 +95,37 @@ class UploadWrittenAuthorisationController @Inject()(
     }
 
   private val maxFileSize = 10485760
+  private val allowedTypes = Set(
+    "application/pdf",
+    "application/msword",
+    "application/vnd.ms-excel",
+    "image/png",
+    "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
+    "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+    "image/jpeg",
+    "text/plain"
+  )
+
   private def valid(file: MultipartFormData.FilePart[TemporaryFile]): Either[String, MultipartFormData.FilePart[TemporaryFile]] = {
-    if (file.ref.file.length() > maxFileSize) {
-      Left(messagesApi("uploadWrittenAuthorisation.error.size"))
-    } else {
-      Right(file)
+
+    val error = validFileSize(file).orElse(validFileType(file))
+    error.fold(Right(file): Either[String, MultipartFormData.FilePart[TemporaryFile]]) {
+      Left(_)
     }
   }
 
+  private def validFileSize(file: MultipartFormData.FilePart[TemporaryFile]): Option[String] = {
```

Comment:
Maybe this would be easier. If we want a more scalable solution other approaches would work better, but I don't think we will need more validations.
` if (file.ref.file.length() > maxFileSize)
      Left ( Max size error Message)
  if (allowedTypes.contains(file.contentType))
     Left  ( Wrong file type Message)

  Right (file)
`